crates/miden-testing/src/kernel_tests/tx/test_faucet.rs

```diff
@@ -529,32 +529,59 @@ fn test_burn_non_fungible_asset_succeeds() {
         begin
             exec.prologue::prepare_transaction
 
-            # add existing non-fungible asset to the vault
+            # add non-fungible asset to the vault
             exec.memory::get_input_vault_root_ptr push.{non_fungible_asset}
             exec.asset_vault::add_non_fungible_asset dropw
 
-            # burn asset
+            # check that the non-fungible asset is in the faucet SMT
+            push.{FAUCET_STORAGE_DATA_SLOT}
+            exec.account::get_item
+            push.{burnt_asset_vault_key}
+            hash
```

Comment:
Can you add a comment that this must be hashed because the vault key is hashed when it is inserted into the faucet SMT?